### PR TITLE
Simplify logic for sessions support detection

### PR DESCRIPTION
**Pull request by mbroadst**
Some standalone servers do support sessions, we should instead depend strictly on the presence of `logicalSessionsTimeoutMinutes` in the server handshake to determine if a server supports sessions.

DRIVERS-1230

### CODE REVIEW

File: source/sessions/driver-sessions.rst [hunk 1]
```diff
@@ -979,19 +977,19 @@ Test Plan
 =========
 
 The test plan SHOULD be run against both replica set and sharded cluster
-topologies.  It MUST NOT be run against a standalone server.
```
Comment (jmikola) [inline source/sessions/driver-sessions.rst:982]: If you're removing the prohibition about running on a standalone, I would venture that the topology really doesn't matter at all. In that case, should the first sentence be changed as well? Perhaps:

> The test plan SHOULD be run against deployments that support sessions

Or just remove this sentence as well.

File: source/sessions/driver-sessions.rst [hunk 2]
```diff
@@ -501,8 +501,6 @@ include a session ID in a ``KILLCURSORS`` command.
 How to Check Whether a Deployment Supports Sessions
 ===================================================
 
-Standalone servers do not support sessions.
-
```
Comment (jmikola) [inline source/sessions/driver-sessions.rst:505]: I think you should also replace "whether a replica set or sharded cluster deployment" with "whether a deployment" if we don't care about topologies anymore.

File: source/sessions/tests/server-support.yml [hunk 3]
```diff
@@ -0,0 +1,102 @@
+runOn:
+  - minServerVersion: "3.6.0"
+    topology: ["single"]
+  - minServerVersion: "4.0"
+    topology: ["replicaset"]
```
Comment (jmikola) [inline source/sessions/tests/server-support.yml:5]: What about these tests requires a 4.0 replica set or 4.1.8 sharded cluster?

We're not using fail points, so this looks like basic session support that would be 3.6+. Let me know if I'm missing something.

File: source/sessions/tests/dirty-session-errors.yml [hunk 4]
```diff
@@ -409,32 +198,3 @@ tests:
       collection:
         data:
           - {_id: 1}
-
-  - description: Dirty implicit session is discarded (non-cursor returning read)
```
Comment (jmikola) [inline source/sessions/tests/dirty-session-errors.yml:413]: Why were these four tests removed?

* Clean explicit session is not discarded
* Clean implicit session is not discarded
* Dirty explicit session is discarded (non-bulk write)
* Dirty implicit session is discarded (non-bulk write)

Were they redundant? They seem unrelated to relaxing the prohibition on standalone deployments.

File: source/sessions/driver-sessions.rst (hunk 1 above)
Comment (mbroadst) [reply source/sessions/driver-sessions.rst:982]: Good idea, I'll just remove it

File: source/sessions/tests/server-support.yml (hunk 3 above)
Comment (mbroadst) [reply source/sessions/tests/server-support.yml:5]: fixed

File: source/sessions/tests/dirty-session-errors.yml (hunk 4 above)
Comment (mbroadst) [reply source/sessions/tests/dirty-session-errors.yml:413]: hm I have no idea what happened here, restoring them now

Comment (mbroadst) [reply source/sessions/tests/dirty-session-errors.yml:413]: This was unintentional, I was copying from a branch that hadn't updated its local tests to include these more recent test additions. Fixed

File: source/sessions/tests/dirty-session-errors.yml [hunk 5]
```diff
@@ -11,96 +11,6 @@ data:
   - {_id: 1}
 
 tests:
-  - description: Clean explicit session is not discarded
```
Comment (jmikola) [inline source/sessions/tests/dirty-session-errors.yml:14]: Two tests are still being removed here:

 * Clean explicit session is not discarded
 * Clean implicit session is not discarded

Is that intentional?

Comment (mbroadst) [reply source/sessions/tests/dirty-session-errors.yml:14]: Yes it was intentional, they were actually just renamed to be the tests in the `server-support` tests
